lib/ember_cli/shell.rb

```diff
@@ -37,6 +37,7 @@ def install
         exec "#{paths.bundler} install"
       end
 
+      exec "#{paths.ember} version || rm -rf #{paths.npm_deps} #{paths.bower_deps}"
```

Comment:
You're absolutely right, catching a `SystemExit` is hacky.

I like your idea of extracting a `#run` method from [`#run!`](https://github.com/thoughtbot/ember-cli-rails/blob/ce5db79326f5aac72615dcb257d10e02bdc1aeb8/lib/ember_cli/runner.rb#L12-L28).

What do you think about:

```rb
def run(command)
  output, status = Open3.capture2e(@env, command, @options)

  @out.write(output)
 
  return status.success?
end

def run!(command)
  exit_successfully = run(command)

  unless exit_successfully
    @err.write <<-MSG.strip_heredoc
      ERROR: Failed command: `#{command}`
      OUTPUT:
        #{output}
    MSG
    exit 1
  end
end
```